src/batchhafnian.hpp

```diff
@@ -117,6 +136,10 @@ inline std::vector<T> hermite_multidimensional(std::vector<T> &R_mat, std::vecto
                 std::transform(jumpFrom.begin(), jumpFrom.end(), prevJump.begin(), tmpjump.begin(), std::minus<int>());
                 ullint prevCoordinate = vec2index(tmpjump, resolution);
                 H[nextCoordinate] = H[nextCoordinate] - (static_cast<T>(jumpFrom[ii]-1))*static_cast<T>(R(k,ii))*H[prevCoordinate];
+				
+				if (renorm) {
+					H[nextCoordinate] = H[nextCoordinate]/renorm_factor(nextPos);
+				}
```

Comment:
Also, you should not be calling a function to calculate factorials everytime. Instead just save the square root factorials all the way to ```resolution``` and just multiply those that you need for each element of the tensor. For this you will need to write a function that allows you to find the indices correspoding to a given element of the tensor inside the vector H. Such function in python is:
```
def find_rep(val, base, n):
    x = np.empty([n]) # In C++ this would be a vector of size n
    local_val = val
    x[0] = 1
    for i in range(1,n):
        x[i] = x[i-1]*base
    digits = np.zeros([n], dtype = int)
    for i in range(n):
        digits[i] = local_val // x[-i-1] # Careful here this is integer division, not float division, although I think C++ does this by default using / 
        local_val = local_val - digits[i]*x[-i-1]

    return digits
```

Reply:
Thanks for the suggestion. Implemented that.